pkg/cmd/pulumi/stack.go

```diff
@@ -196,6 +99,107 @@ func newStackCmd() *cobra.Command {
 	return cmd
 }
 
+func runStack(ctx context.Context, s backend.Stack, out io.Writer, args stackArgs) error {
+	if args.showStackName {
+		fmt.Fprintln(out, s.Ref().String())
```

Comment:
Isn't this going to change the behavior when `--fully-qualify-stack-names` isn't set? Previously, it'd always show only the name of the stack. Now this may show the owner and project as well, depending on whether those are available, per:

https://github.com/pulumi/pulumi/blob/0cd3a460c366f2afaccc6d51819a6492e5363964/pkg/backend/httpstate/stack.go#L54-L84

and:

https://github.com/pulumi/pulumi/blob/0cd3a460c366f2afaccc6d51819a6492e5363964/pkg/backend/diy/backend.go#L138-L162

Arguably, that may be "better" behavior. But it is a behavior change if someone is expecting just the name itself, right?

Reply:
Good point. My take was that this behavior is indeed better and more consistent, but thinking about it again - it may break users. Also, I guess one could read the "stack --show-name" as show just the stack name without anything else.
I went ahead and restored the old behavior for the non-full case.